### PR TITLE
Prep 2.0.0

### PR DESCRIPTION
**Pull request by woodruffw**
~~Draft until we're ready to merge this.~~

### CODE REVIEW

File: sigstore/transparency.py [hunk 1]
```diff
@@ -24,9 +24,9 @@
     BaseModel,
     ConfigDict,
     Field,
-    FieldValidationInfo,
     StrictInt,
     StrictStr,
+    ValidationInfo,
```
Comment (woodruffw) [inline sigstore/transparency.py:29]: I haven't root caused this, but this looks like a semver-breaking change in Pydantic.

Comment (woodruffw) [reply sigstore/transparency.py:29]: Hmm, more borkage with Pydantic: https://github.com/sigstore-conformance/extremely-dangerous-public-oidc-beacon/actions/runs/6315666569/job/17148672809

Comment (woodruffw) [reply sigstore/transparency.py:29]: https://github.com/sigstore-conformance/extremely-dangerous-public-oidc-beacon/pull/6

Comment (jku) [reply sigstore/transparency.py:29]: > I haven't root caused this, but this looks like a semver-breaking change in Pydantic.

Either that or I used supposedly non-public API by accident... https://github.com/pydantic/pydantic/pull/7542 happened last friday

Comment (woodruffw) [reply sigstore/transparency.py:29]: Ah yep, could be. It looks like `FieldValidationInfo` might have been deprecated at some point (prior to removal), but I can't find a good reference for that (or for why).

**Comment by woodruffw** [comment]
This should be safe to merge whenever; we can create the release anytime after it's merged.

**Review by jku — APPROVED** [review]
LGTM.

For reference on that pydantic issue
* I believe FieldValidationInfo still works but the way they marked it deprecated (which seems to be what happened last friday) is what mypy flags as suspicious
* the fix here looks right to me

**Review by di — APPROVED** [review]

**Pull request merged**